[ovn] Update backup/restore procedure (#1396)
openstack-k8s-operators/data-plane-adoption: docs_user/modules/proc_migrating-ovn-data.adoc

* The `OpenStackControlPlane` resource is created.
* `NetworkAttachmentDefinition` custom resources (CRs) for the original cluster are defined. Specifically, the `internalapi` network is defined.
* The original {networking_first_ref} and OVN `northd` are not running.
* The cloud is migrated to the Modular Layer 2 plug-in with Open Virtual Networking (ML2/OVN) mechanism driver.

> * SSH access is available to the original {rhos_prev_long} Controller nodes for direct database file access.

Reviewer: ```suggestion
* SSH access is available to the original {rhos_prev_long} Controller nodes to allow direct access to database files.
```

Author: Done

STORAGE_CLASS=local-storage
OVSDB_IMAGE=registry.redhat.io/rhoso/openstack-ovn-base-rhel9:18.0
endif::[]
----
+
ifeval::["{build_variant}" == "ospdo"]
[NOTE]
If you use a disconnected environment director Operator deployment, use
`OVSDB_IMAGE=registry.redhat.io/rhoso/openstack-ovn-base-rhel9@sha256:967046c6bdb8f55c236085b5c5f9667f0dbb9f3ac52a6560dd36a6bfac051e1f`.
For more information, see link:https://docs.redhat.com/en/documentation/red_hat_openstack_platform/17.1/html-single/deploying_an_overcloud_in_a_red_hat_openshift_container_platform_cluster_with_director_operator/index#proc_configuring-an-airgapped-environment_air-gapped-environment[Configuring an airgapped environment] in _Deploying an overcloud in a Red Hat OpenShift Container Platform cluster with director Operator_.
endif::[]

> * Ensure you have SSH access to the original {rhos_prev_long} Controller nodes. This procedure extracts database files directly from the controller filesystem.

Reviewer: ```suggestion
* Ensure that you have SSH access to the original {rhos_prev_long} Controller nodes to extract database files directly from the Controller file system.
```

Author: Done

+
* Replace `<ospdo_namespace>` with your OSPdO namespace.
endif::[]
. Convert the OVN cluster databases to standalone format on the Controller node and copy them to the local system:
+
----
# Set up variables for the controller SSH connection
CONTROLLER1_SSH="ssh -i ~/.ssh/id_rsa heat-admin@$CONTROLLER_IP"

# Create working directory on controller and convert databases
$CONTROLLER1_SSH sudo mkdir -p /tmp/ovn_standalone_conversion
$CONTROLLER1_SSH sudo ovsdb-tool cluster-to-standalone /tmp/ovn_standalone_conversion/ovnnb_db.db /var/lib/openvswitch/ovn/ovnnb_db.db
$CONTROLLER1_SSH sudo ovsdb-tool cluster-to-standalone /tmp/ovn_standalone_conversion/ovnsb_db.db /var/lib/openvswitch/ovn/ovnsb_db.db
$CONTROLLER1_SSH sudo chmod 644 /tmp/ovn_standalone_conversion/ovn*.db

# Create local directory and copy converted databases

> mkdir -p /tmp/ovn_adoption_dbs

Reviewer 2: Just noting the new requirement of local temporary storage on the host driving the adoption. We already have the control plane YAML files there so it's not the first local file requirement. And it's probably better than the alternatives like giving some pod a permission to ssh into the controllers. So +1.

CONTROLLER1_SCP=$(echo "$CONTROLLER1_SSH" | sed 's/^ssh/scp/')
${CONTROLLER1_SCP}:/tmp/ovn_standalone_conversion/ovnnb_db.db /tmp/ovn_adoption_dbs/
${CONTROLLER1_SCP}:/tmp/ovn_standalone_conversion/ovnsb_db.db /tmp/ovn_adoption_dbs/

# Cleanup on controller
$CONTROLLER1_SSH sudo rm -rf /tmp/ovn_standalone_conversion
----
+

> . Create the PersistentVolumeClaim resources that will be used by the OVN database pods:

Reviewer: ```suggestion
. Create the `PersistentVolumeClaim` resources that will be used by the OVN database pods:
```

          dbType: SB
          storageRequest: 10G
          networkAttachment: internalapi
'
----
+
[NOTE]
The OVN database pods will automatically handle schema conversion during startup and form a proper 3-replica cluster with the preserved data.

. Wait for the OVN database services to start up:
+
----
$ oc wait pod --for condition=Ready --selector=service=ovsdbserver-nb --timeout=300s
$ oc wait pod --for condition=Ready --selector=service=ovsdbserver-sb --timeout=300s
----

> . Verify that the data was successfully migrated

Reviewer: ```suggestion
. Verify that the data was successfully migrated:
```

Author: Done

+
----
$ oc exec -it ovsdbserver-nb-0 -- ovn-nbctl show
$ oc exec -it ovsdbserver-sb-0 -- ovn-sbctl show
$ oc exec -it ovsdbserver-sb-0 -- ovn-sbctl list Chassis
----

> +

Reviewer: ```suggestion
```

Author: Done

[NOTE]
Running OVN gateways on {OpenShiftShort} nodes might be prone to data plane downtime during Open vSwitch upgrades. Consider running OVN gateways on dedicated `Networker` data plane nodes for production deployments instead.

. The migration is now complete.

> Consider creating a backup of the current state for rollback purposes:

Reviewer: ```suggestion
```
I'm not sure we need this sentence. I think the note below suffices, but let me know if you disagree.

Author: Removed it

+
[NOTE]
The OVN databases now contain the migrated data with preserved UUIDs. Consider taking a snapshot of the database PVCs for backup purposes. For more information, see link:{defaultOCPURL}/storage/index#lvms-about-volume-snapsot_logical-volume-manager-storage[About volume snapshots] in _OpenShift Container Platform storage overview_.

> . Stop the adopted OVN database servers:

Reviewer: Is this step still okay where it is? The step above now says that the migration is complete, but customers still need to stop the adopted OVN database servers?

Author: yes we still need to stop the original OVN database servers after adoptioin
